draft-birkholz-rats-corim.md

```diff
@@ -132,17 +353,436 @@ concise-mid-tag = {
+
+~~~~ CDDL
+environment-map = non-empty<{
+  ? comid.class => class-map
+  ? comid.instance => $instance-id-type-choice
+  ? comid.group => $group-id-type-choice
+}>
+
+$instance-id-type-choice /= tagged-ueid-type
+$instance-id-type-choice /= tagged-uuid-type
+
+$group-id-type-choice /= tagged-uuid-type
+~~~~
+
+The following describes each member of the environment-map container.
+
+comid-class:
+
+: A composite identifier for classes of environments/modules.
+
+comid.instance:
+
+: An identifier for distinct instances of environments/modules that is either a UEID or a UUID.
+
+comid.group:
+
+: An identifier for a group of environments/modules that is a UUID.
+
+## The class-map Container
+
+This map enables a composite identifier intended to uniquely identify modules that are of a distinct class of devices. Effectively, all provided members in combination are a composite module class identifier.  This rule and its constraints MUST be followed when generating or validating a CoMID tag. This rule and its constraints MUST be followed when generating or validating a CoMID tag.
+
+~~~~ CDDL
+class-map = non-empty<{
+  ; TODO(tho) add text "class-map SHOULD include class-id"
+  ? comid.class-id => $class-id-type-choice
+  ? comid.vendor => tstr
+  ? comid.model => tstr
+  ? comid.layer => uint
+  ? comid.index => uint
+}>
+
+$class-id-type-choice /= tagged-oid-type
+$class-id-type-choice /= tagged-impl-id-type
+$class-id-type-choice /= tagged-uuid-type
+~~~~
+
+The following describes each member of the class-map container.
+
+{: #model-measurement-values-map}
+## The measurement-map and measurement-values-map Containers
+
+One of the targets (range) that a triple-map can point to in order to associate it with a module (domain) is the measurement-map. This map is used to provide reference measurements values that can be compared with Evidence Claim values or Endorsements and endorsed values from other sources than the corresponding CoRIM. `measurement-map` comes with a measurement key that identifies the measured element with via a OID reference or a UUID. `measurement-values-map` contains the actual measurements associated with the module(s). Byte strings with corresponding bit masks that highlights which bits in the byte string are used as reference measurements or endorsement are located in `raw-value-group`. The members of `measurement-values-map` provide well-defined and well-scoped semantics for reference measurement or endorsements with respect to a given module instance, class, or group. This rule and its constraints MUST be followed when generating or validating a CoMID tag.
```

Comment:
Considering the length of statement, request please consider simple split on the following lines:

This map is used to provide (a) reference measurements values that can be compared with Evidence Claim values - reported from other sources, example attesting device.
AND/OR (b) Endorsements that can be compared with endorsed values received from other sources than the corresponding CoRIM.